mautic/mautic
# It is necessary to know more than just the ID to download an asset

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:


Currently it is enough to guess the ID and the asset will download. E.g.: https://mautic.ddev.site/asset/1:. In this case all assets can be downloaded via IDs. This PR suggests the solution when the user has to know the full path. Otherweise a 404 is displayed: https://mautic.ddev.site/asset/1:alias. 

https://aivie.ddev.site/asset/53:my-download-alias
#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new asset.
3. Try to download the asset by different ways. E.g. https://mautic.ddev.site/asset/1:, https://mautic.ddev.site/asset/1, https://mautic.ddev.site/asset/1:alias. Only path with the alias works. 404er is displayed in other cases.
